config/routes.rb

```diff
@@ -1,56 +1,44 @@
 Rails.application.routes.draw do
-  # The priority is based upon order of creation: first created -> highest priority.
-  # See how all your routes lay out with "rake routes".
+  root 'users#index'
+
+  resources :vendors do
+  	resources :products do
+	  	resources :sales, only: [:create, :new]
+  	end
+  	collection do
+	  	get :show_vendors
+	  end
+  	member do 
+  		resources :sales, only: [:index]
+  	end
+
+  	collection do 
+	  	get :search
+	  end
+  end
+
+  resources :users do
+  	collection do
+  		get 'markets'
+  		get 'markets/:id', action: "show_market", as: "show_market"
+  	end
+  end
+
+  resources :markets do
+  	get 'show_vendors', action: "show_vendors", as: "by_vendor"
+  	get 'new_vendor/', action: "new_vendor", as: "new_vendor"
+  	post 'create_vendor/', action: "create_vendor", as: "create_vendor"
```

Comment:
Try to be consistent about whether or not you put the `/` at the end of the route